### PR TITLE
Forbid URLs to be inserted into Name fields

### LINKED ISSUE #13524: Problem with customer account creation
**Describe the bug**

**To Reproduce**

**Screenshots**
If applicable, add screenshots or screenrecords to help explain your problem.

**Additionnal information**
PrestaShop version: N/A
PHP version: N/A

### PR DESCRIPTION
**Pull request by PierreRambaud**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.6.1.x
| Description?  | Only forbid for customer name and not everywhere
| Type?         | bug fix
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #13524 
| How to test?  | Attempt, from front-office, to create a customer with an URL as first name or last name. Now it is not possible anymore.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/13607)

### CODE REVIEW

File: js/validate.js [hunk 1]
```diff
@@ -75,33 +75,39 @@ var unicode_hack = (function() {
...
-* versions in the future. If you wish to customize PrestaShop for your
-* needs please refer to http://www.prestashop.com for more information.
-*
-*  @author PrestaShop SA <contact@prestashop.com>
-*  @copyright  2007-2017 PrestaShop SA
-*  @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
-*  International Registered Trademark & Property of PrestaShop SA
-*/
+/**
+ * 2007-2017 PrestaShop
```
Comment (matks) [inline js/validate.js:79]: Shouldn't we update the license header then ?

File: classes/Validate.php [hunk 2]
```diff
@@ -148,6 +148,22 @@ public static function isImageSize($size)
...
     }
 
+    /**
+     * Check whether given customer name is valid
+     *
+     * @param string $name Name to validate
+     *
+     * @return int 1 if given input is a name, 0 else
+     */
+    public static function isCustomerName($name)
```
Comment (matks) [inline classes/Validate.php:158]: I did not think of it first but ... should'nt we update the license header to `2007-2019 PrestaShop` then ?

File: themes/default-bootstrap/authentication.tpl [hunk 3]
```diff
@@ -444,11 +444,11 @@
...
+				<input onkeyup="$('#firstname').val(this.value);" type="text" class="is_required validate form-control" data-validate="isCustomerName" id="customer_firstname" name="customer_firstname" value="{if isset($smarty.post.customer_firstname)}{$smarty.post.customer_firstname}{/if}" />
 			</div>
 			<div class="required form-group">
 				<label for="customer_lastname">{l s='Last name'} <sup>*</sup></label>
-				<input onkeyup="$('#lastname').val(this.value);" type="text" class="is_required validate form-control" data-validate="isName" id="customer_lastname" name="customer_lastname" value="{if isset($smarty.post.customer_lastname)}{$smarty.post.customer_lastname}{/if}" />
+				<input onkeyup="$('#lastname').val(this.value);" type="text" class="is_required validate form-control" data-validate="isCustomerName" id="customer_lastname" name="customer_lastname" value="{if isset($smarty.post.customer_lastname)}{$smarty.post.customer_lastname}{/if}" />
```
Comment (matks) [inline themes/default-bootstrap/authentication.tpl:451]: 👏

File: js/validate.js (hunk 1 above)
Comment (PierreRambaud) [reply js/validate.js:79]: Don't really know, because we don't do it for the old release, but could be done in a new PR

Comment (matks) [reply js/validate.js:79]: Well we never asked ourselves the question, but maybe it's time to do it 😄

File: tests/Unit/classes/ValidateCoreTest.php [hunk 4]
```diff
@@ -213,7 +258,7 @@ public function isUnsignedFloatDataProvider()
             $this->trueFloatDataProvider(),
             array(
                 array(false, -12.2151),
-                array(false, -12,2151),
+                array(false, -12, 2151),
```
Comment (eternoendless) [inline tests/Unit/classes/ValidateCoreTest.php:261]: This test is weird

**Review by eternoendless — APPROVED** [review]

File: tests/Unit/classes/ValidateCoreTest.php (hunk 4 above)
Comment (PierreRambaud) [reply tests/Unit/classes/ValidateCoreTest.php:261]: Was an old test, cs-fixer just add a space here :D

Comment (PierreRambaud) [reply tests/Unit/classes/ValidateCoreTest.php:261]: But yes, it's weird :/

**Comment by eternoendless** [comment]
Thank you @PierreRambaud

File: classes/Validate.php (hunk 2 above)
Comment (amir-dragon) [reply classes/Validate.php:158]: yes we might need to :)

**Pull request merged**